### PR TITLE
47 modals validation

### PR DESCRIPTION
**Pull request by rkzel**
This is working example of validation done on BasicInfo panel. If it's ok I'll do the same to the rest of them.

### CODE REVIEW

**Comment by rkzel** [comment]
Okay, this should work.

**Comment by rkzel** [comment]
Issues solved: 47 and 48.

File: app/src/components/modals/WorkHistory.js [hunk 1]
```diff
@@ -1,19 +1,70 @@
...
+
+const validateJobTitle = validator.compile({
+  type: 'string',
+  minLength: 1,
+  maxLength: 64,
+})
+
+const validateDescription = validator.compile({
+  type: 'string',
+  minLength: 10,
```
Comment (Schwartz10) [inline app/src/components/modals/WorkHistory.js]: Description is an optional field, whereas start and end date are not: 

https://docs.google.com/document/d/1Vluc0R8qEgtxv8m2GpC6a1KqAfZ2sqLGCJGJMcTRiV4/edit#heading=h.7ovnn4tlfkts

File: app/src/components/modals/BasicInformation.js [hunk 2]
```diff
@@ -42,15 +73,16 @@ const BasicInformation = ({
...
           wide
           value={getFormValue('website')}
           onChange={e => onChange(e.target.value, 'website')}
           type="url"
+          error={validationErrors['website']}
         />
       </div>
 
-      <Button mode="strong" wide onClick={() => saveProfile(ethereumAddress)}>
+      <Button mode="strong" wide onClick={() => validateAndSave()}>
```
[the remark this answers is not in the document]
Comment (rkzel) [reply app/src/components/modals/BasicInformation.js]: Oh right, I probably got too used to those functions, and overused them in some places. Fixed.

File: app/src/components/modals/EducationHistory.js [hunk 3]
```diff
@@ -113,7 +139,7 @@ const EducationHistory = ({
         type="educationHistory"
       />
 
-      <Button wide mode="strong" onClick={() => saveProfile(ethereumAddress)}>
+      <Button wide mode="strong" onClick={() => validateAndSave()}>
```
[the remark this answers is not in the document]
Comment (rkzel) [reply app/src/components/modals/EducationHistory.js]: Fixed.

File: app/src/components/modals/WorkHistory.js [hunk 4]
```diff
@@ -1,19 +1,70 @@
...
 import DateDropdowns from '../DateDropdowns'
-import { Label } from '../styled-components'
+import {
+  Label,
+  TextInputWithValidation,
+  TextMultilineWithValidation,
+} from '../styled-components'
+import validator from '../../utils/validation'
+
+const validateWorkPlace = validator.compile({
```
[the remark this answers is not in the document]
Comment (rkzel) [reply app/src/components/modals/WorkHistory.js]: Done.

File: app/src/components/modals/EducationHistory.js [hunk 5]
```diff
@@ -1,19 +1,42 @@
...
-import { ModalWrapper, TwoColumnsRow } from './ModalWrapper'
+import { ModalWrapper, TwoColumnsRow, DisplayErrors } from './ModalWrapper'
 import { useDate } from '../../hooks'
 import { years } from '../../utils'
 import DateDropdowns from '../DateDropdowns'
-import { Label } from '../styled-components'
+import { Label, TextInputWithValidation } from '../styled-components'
+import validator from '../../utils/validation'
+
+const validateOrganization = validator.compile({
```
[the remark this answers is not in the document]
Comment (rkzel) [reply app/src/components/modals/EducationHistory.js]: Done, and renamed...

File: app/src/components/modals/BasicInformation.js [hunk 6]
```diff
@@ -1,24 +1,55 @@
...
+import React, { useState } from 'react'
 import { Button, TextInput } from '@aragon/ui'
 import PropTypes from 'prop-types'
-import { ModalWrapper, TwoColumnsRow } from './ModalWrapper'
-import { Label } from '../styled-components'
+import { ModalWrapper, TwoColumnsRow, DisplayErrors } from './ModalWrapper'
+import { Label, TextInputWithValidation } from '../styled-components'
+import validator from '../../utils/validation'
+
+const validateName = validator.compile({
```
[the remark this answers is not in the document]
Comment (rkzel) [reply app/src/components/modals/BasicInformation.js]: Done.

File: app/src/components/modals/index.js [hunk 7]
```diff
@@ -127,10 +120,32 @@ const UserInfoModal = ({ ethereumAddress }) => {
...
+    !boxes[ethereumAddress].savedProfileSucess
+      ? { error: `Error saving profile: ${error.message}` }
+      : {}
+  const removingError =
+    boxes[ethereumAddress].removedItem &&
+    !boxes[ethereumAddress].removedItemSuccess
+      ? { error: `Error removing item: ${error.message}` }
+      : {}
+
+  const props = {
```
[the remark this answers is not in the document]
Comment (rkzel) [reply app/src/components/modals/index.js]: modalsCommonProps seemed appropriate.

File: app/src/stateManagers/box/states.js [hunk 8]
```diff
@@ -196,7 +196,7 @@ export const requestedSaveProfileError = (state, error) => ({
   ...state,
   savingProfile: false,
   savedProfile: true,
-  savedProfileSucess: true,
+  savedProfileSucess: false,
```
[the remark this answers is not in the document]
Comment (rkzel) [reply app/src/stateManagers/box/states.js]: :)

File: app/src/utils/validation.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+const validator = new JsonSchemaValidator({
+  coerceTypes: true,
+  useDefaults: true,
+})
+
+validator.addFormat('address', {
+  type: 'string',
+  validate: isAddress,
+})
+
```
[the remark this answers is not in the document]
Comment (rkzel) [reply app/src/utils/validation.js]: Done.

File: app/src/components/modals/WorkHistory.js (hunk 1 above)
Comment (rkzel) [reply app/src/components/modals/WorkHistory.js]: Done. Works differently in Edu and Work - dates are only required in Work. One thing I can't get to work is highlighting border when dates are not correct - borders must be hardcoded somewhere in Aragon components. But error message is clear enough...

**Pull request merged**